Mathlib/Algebra/Group/Fin.lean

```diff
@@ -123,9 +123,55 @@ lemma sub_one_lt_iff {k : Fin (n + 1)} : k - 1 < k ↔ 0 < k :=
   not_iff_not.1 <| by simp only [lt_def, not_lt, val_fin_le, le_sub_one_iff, le_zero_iff]
 #align fin.sub_one_lt_iff Fin.sub_one_lt_iff
 
+lemma lt_sub_iff {n : ℕ} {a b : Fin n} : a < a - b ↔ a < b := by
+  cases' n with n
+  · exact a.elim0
+  by_cases h: a < b
```

Comment:
```suggestion
  by_cases h : a < b
```
but `constructor <;> intro h` might be more typical

Reply:
Rearranged to use constructor